### PR TITLE
Reviving corals with water bottles

### PR DESCRIPTION
**Pull request by Nameless-123**
By: Ghoulboy (that name was taken)
This is a script with enables the making of a coral block farm with carpet mod's `renewableCorals` set to `true`

### CODE REVIEW

File: programs/survival/revive_coral [hunk 1]
```diff
@@ -0,0 +1,16 @@
+//allows player to revive dead corals with water bottles
+__on_player_right_clicks_block (player, item_tuple,hand, block,face,hitvec) -> (
+	l(item, count, nbt) = item_tuple || l('None', 0, null);
+	dead=l('dead_brain_coral_block','dead_bubble_coral_block','dead_fire_coral_block','dead_horn_coral_block','dead_tube_coral_block');
```
Comment (gnembon) [inline programs/survival/revive_coral]: you could just detect if blocks matches 'dead_\w+_coral_block` and remove 'dead_' from the block name when setting. A suggestion

File: programs/survival/revive_coral [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+	live=l('brain_coral_block','bubble_coral_block','fire_coral_block','horn_coral_block','tube_coral_block');
+	listpos=dead ~ block;
+	if(item=='potion'&& dead ~ block !=null,
```
Comment (gnembon) [inline programs/survival/revive_coral]: is 'potion' also applying to other potions, not just water bottles?

File: programs/survival/revive_coral [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+		print('WOrks');
```
Comment (gnembon) [inline programs/survival/revive_coral]: Now, we know it WOrks, you can remove it :)

File: programs/survival/revive_coral (hunk 1 above)
Comment (Nameless-123) [reply programs/survival/revive_coral]: I have no clue how to do, could you please explain further?

File: programs/survival/revive_coral (hunk 3 above)
Comment (Nameless-123) [reply programs/survival/revive_coral]: o crap I forgot thanks

**Comment by Nameless-123** [comment]
Sorry, this is like my first function scarpet script, and im new to programming, so I made a ton of stupid errors

File: programs/survival/revive_coral (hunk 2 above)
Comment (Nameless-123) [reply programs/survival/revive_coral]: yeah, true

**Comment by Nameless-123** [comment]
Also, why does the '_on_player_right_clicks_block' method need all the parameters for it to work? I mean, I only need the 'player', 'item_tuple' and 'block' parameters, so why do I need the rest?

File: programs/survival/revive_coral (hunk 1 above)
Comment (Nameless-123) [reply programs/survival/revive_coral]: never mind, it took 15 mins of testing to figure out

**Comment by gnembon** [comment]
the rest is there in case you need it. if you don't need it don't use it. Some people may need those that's why.

**Comment by Nameless-123** [comment]
I get that, but I don't see why it doesn't work if I don't use params, unless its easier to code that way, in which case that's all the explanation I need

File: programs/survival/revive_coral [hunk 4]
```diff
@@ -0,0 +1,14 @@
+//allows player to revive dead corals with water bottles
+__on_player_right_clicks_block (player, item_tuple,hand, block,face,hitvec) -> (
+	l(item, count, nbt) = item_tuple || l('None', 0, null);
+	position=pos(block);
+	types=l('brain','bubble','fire','horn','tube');
+	block=block-'dead_'-'_coral_block';
+	if(item=='potion'&& types ~ block != null && nbt=='{Potion:"minecraft:water"}',
```
Comment (gnembon) [inline programs/survival/revive_coral]: that would work if that's all what's in the nbt tag. You could make it more robust by using `nbt:'Potion' == 'minecraft:water'`

**Comment by gnembon** [comment]
when the function has incorrect number of arguments, it is not plugged in to handle that event, since scarpet makes sure the function is correct and has correct argument count.

File: programs/survival/revive_coral (hunk 4 above)
Comment (Nameless-123) [reply programs/survival/revive_coral]: thx I will use

**Pull request merged**